### PR TITLE
Trace function signature types

### LINKED ISSUE #291: Bindings for types mentioned in whitelisted function's signature are not generated
The transitive nature of whitelisting means that bindings for these types *should* be generated.

Testcase incoming...

### PR DESCRIPTION
**Pull request by fitzgen**
This extends the `TypeCollector` trait implementation for `Item` to
consider items of kind `Function` and to collect the types found in the
function's signature.

Fixes #291 

r? @emilio

### CODE REVIEW

File: libbindgen/tests/expectations/tests/type-referenced-by-whitelisted-function.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* automatically generated by rust-bindgen */
+
+
+#![allow(non_snake_case)]
+
+
+
```
Comment (fitzgen) [inline libbindgen/tests/expectations/tests/type-referenced-by-whitelisted-function.rs]: er wait woops

**Comment by fitzgen** [comment]
Fixed bad test case.

File: libbindgen/tests/expectations/tests/type-referenced-by-whitelisted-function.rs (hunk 1 above)
Comment (emilio) [inline libbindgen/tests/expectations/tests/type-referenced-by-whitelisted-function.rs]: Heh

**Comment by emilio** [comment]
@bors-servo r+

Thanks!

**Pull request merged**